app/src/main/java/com/example/demomvp/data/source/remote/fetchJson/GetDataFromUrl.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.demomvp.data.source.remote.fetchJson
+
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.lang.Exception
+import java.net.HttpURLConnection
+import java.net.URL
+
+class GetDataFromUrl {
+    fun getDataFromUrl(url: String): String? {
+        var urlConn: HttpURLConnection? = null
+        var result = ""
+        try {
+            urlConn = URL(url).openConnection() as HttpURLConnection
+            if (urlConn != null) {
+                val code = urlConn!!.responseCode
+                if (code != 200) {
+                    throw IOException("Invalid response from server: $code")
+                }
+                val rd = BufferedReader(InputStreamReader(urlConn!!.inputStream))
+                var check: Boolean = true
```

Comment:
biến này dùng làm gì thế e?
```suggestion
                var isCheck = true
```

Reply:
var isCheck = true
while (true) {
    var line = rd.readLine()
    if (line == null) {
        break
    } else {
        if (isCheck){
            result += line
            isCheck = false
        }
        else {
            result += "\n" +  line
        }
    }
}


đoạn này em đọc từng dòng từ BufferedReader nên em dùng biến trên để dòng đầu tiên k ghi ký hiệu "\n" vào, chỉ ở những dòng tiếp theo mới thêm "\n"